packages/next/client/index.tsx

```diff
@@ -536,7 +535,9 @@ function Root({
   // We should ask to measure the Web Vitals after rendering completes so we
   // don't cause any hydration delay:
   React.useEffect(() => {
-    measureWebVitals(onPerfEntry)
+    if (process.env.__NEXT_ANALYTICS_ID) {
```

Comment:
This is also needed when `reportWebVitals` is being used in `_app` which isn't known before build so isn't able to be excluded in this way.

Reply:
Ah, I missed that. Good catch.

I still think though that there should be a way to exclude `web-vitals`.

What if I add a flag in the Next.js config, perhaps `removeWebVitals`?

Reply:
Could potentially do a pre-check and see if a custom `_app` is present and if it contains `reportWebVitals` before starting the webpack compilation and if it doesn't contain it or doesn't have a custom app a `DefineEnv` value can be set to signal it can be excluded.